bundles/org.eclipse.equinox.preferences/src/org/eclipse/core/internal/preferences/Activator.java

```diff
@@ -75,6 +80,33 @@ public void start(BundleContext context) throws Exception {
 		// use the string for the class name here in case the registry isn't around
 		registryServiceTracker = new ServiceTracker<>(bundleContext, "org.eclipse.core.runtime.IExtensionRegistry", this); //$NON-NLS-1$
 		registryServiceTracker.open();
+		locationTracker = new ServiceTracker<>(context,
```

Comment:
I've noticed the framework is not optimized for `ServiceTracker.open` when this constructor is used with supplied filter.  I opened https://github.com/eclipse-equinox/equinox.framework/issues/45 to improve that.

Reply:
I assume this will be improved so so there is nothing to do here?

Reply:
Yes, that was just meant as an FYI.  I'm somewhat surprised this hasn't shown up in anyone's performance profiling.  With that said, I'm not that optimistic my change will actually result in that much improvement, but I plan to move forward because that was the intended usage by the framework of the internal `FilterImpl` class.